### PR TITLE
feature/separate queues for synthesis

### PR DESCRIPTION
**Pull request by sywhb**
- Add text-to-speech queue name and location to env
- Use text-to-speech queue
- Add the priority to request param and use the correct queue accordingly
GET `/:id/:outputFormat/:priority/:voice?`

### CODE REVIEW

File: packages/api/src/util.ts [hunk 1]
```diff
@@ -233,6 +235,8 @@ export function getEnv(): BackendEnv {
     reminderTaskHanderUrl: parse('REMINDER_TASK_HANDLER_URL'),
     integrationTaskHandlerUrl: parse('INTEGRATION_TASK_HANDLER_URL'),
     textToSpeechTaskHandlerUrl: parse('TEXT_TO_SPEECH_TASK_HANDLER_URL'),
+    textToSpeechName: parse('TEXT_TO_SPEECH_QUEUE_NAME'),
```
Comment (jacksonh) [inline packages/api/src/util.ts:238]: I wonder if we can just hardcode these queue names. Doesn't seem like we will ever want to dynamically change the names

File: packages/api/src/util.ts [hunk 2]
```diff
@@ -233,6 +235,8 @@ export function getEnv(): BackendEnv {
...
+    textToSpeechLocation: parse('TEXT_TO_SPEECH_QUEUE_LOCATION'),
```
Comment (jacksonh) [inline packages/api/src/util.ts:239]: I also wonder if we can get this from our current location, or if the API defaults to that. We always want all infra in the same location, so maybe we don't need to specify.

File: packages/api/src/utils/createTask.ts [hunk 3]
```diff
@@ -240,10 +236,17 @@ export const enqueueParseRequest = async (
     return ''
   }
 
+  // use GCF url for low priority tasks
+  const taskHandlerUrl =
```
Comment (jacksonh) [inline packages/api/src/utils/createTask.ts:240]: Hard to tell from the diff, would it be cleaner if content fetch also passed the queue name in?

File: packages/api/src/util.ts (hunk 2 above)
Comment (jacksonh) [reply packages/api/src/util.ts:239]: I looked at what google provides and I guess not. But maybe we could have a single `GCP_LOCATION`

Comment (sywhb) [reply packages/api/src/util.ts:239]: Haha. Yes, I also just looked at `cloud_tasks_client.ts`...
Yeah, I agree to hardcode these configs. I was thinking about having all these configs as env var is for self-hosting.

File: packages/api/src/utils/createTask.ts (hunk 3 above)
Comment (sywhb) [reply packages/api/src/utils/createTask.ts:240]: Yes, let me change this

File: packages/api/src/util.ts (hunk 2 above)
Comment (jacksonh) [reply packages/api/src/util.ts]: yeah no real need to have lots of secrets.

**Review by jacksonh — APPROVED** [review]

File: packages/api/src/util.ts (hunk 2 above)
Comment (sywhb) [reply packages/api/src/util.ts]: Thanks!

**Pull request merged**